### PR TITLE
Add context menu component

### PR DESCRIPTION
**Pull request by tristen**
This PR:

- Updates Radix packages to their latest.
- Adds a context menu component powered by the [API provided in Radix](https://www.radix-ui.com/docs/primitives/components/context-menu#api-reference).

### CODE REVIEW

**Comment by tristen** [comment]
Just flagging that the diff of changes looks big on account of snapshots changing with radix package updates. The implementation of the context menu + tests is like half of it.

File: src/components/context-menu/context-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+type Options = {
+  title?: string;
+  options: Array<Option>;
+}
+
+type Option = {
+  label: ReactNode;
+  value: string;
+  keyCode?: string;
```
Comment (VLuisa) [inline src/components/context-menu/context-menu.tsx:15]: Would we expect someone to use the `<KeybindingTooltipWrapper />` with this in Studio? Or is that redundant?

File: src/components/context-menu/context-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  themeItem?: string;
+  themeTitle?: string;
+  themeSeparator?: string;
+}
+
+export default function ContextMenu({
+  children,
+  onChange,
+  options,
+  themeControlWrapper = 'bg-white round shadow-darken10 px6 py6 txt-s wmin240',
```
Comment (VLuisa) [inline src/components/context-menu/context-menu.tsx:34]: Nitpick: For the default values, the `shadow-darken10` feels odd on the disabled line item b/c it hints at interactivity that's not available

File: src/components/context-menu/context-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  themeItem = 'px6 py6 round color-blue-on-hover bg-gray-lighter-on-hover',
+  themeTitle = 'px6 py6 txt-bold',
+  themeSeparator = 'border-t border--gray-light my6'
```
Comment (VLuisa) [inline src/components/context-menu/context-menu.tsx:37]: having this in the api is nice 👍

File: src/components/context-menu/context-menu.tsx (hunk 2 above)
Comment (VLuisa) [reply src/components/context-menu/context-menu.tsx:34]: `txt-s` felt a bit small for the list item labels, what do you think about `txt-ms` instead?

File: src/docs/index.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './app';
 import './prism-theme.css';
+import './miscellaneous.css';
```
Comment (VLuisa) [inline src/docs/index.js:5]: What's this for?

Comment (VLuisa) [reply src/docs/index.js:5]: Oh duh, scrolling down to the actual file 😅

File: src/components/context-menu/context-menu.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+ContextMenu.propTypes = {
+  /**
+   * The trigger element.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * An array of objects that represent each option group.
+   * Each object contain two fields:
```
Comment (VLuisa) [inline src/components/context-menu/context-menu.tsx:113]:
```suggestion
   * Each object contains two fields:
```
typo

**Review by VLuisa — CHANGES_REQUESTED** [review]
This is really exciting! I think the overall style looks great for use in Studio and I like the flexibility for styling you're providing in the api. Left a few small comments for your thoughts.

File: src/components/context-menu/context-menu.tsx (hunk 1 above)
Comment (tristen) [reply src/components/context-menu/context-menu.tsx:15]: ohhh that's a great call! so instead os a string this should accept a `ReactNode`?

File: src/docs/index.js (hunk 4 above)
Comment (tristen) [reply src/docs/index.js:5]: It is ... It _might_ actually be worth adding these specific rules to mbx-assembly instead 🤔

File: src/components/context-menu/context-menu.tsx (hunk 2 above)
Comment (tristen) [reply src/components/context-menu/context-menu.tsx:34]: > Nitpick: For the default values, the `shadow-darken10` feels odd on the disabled line item b/c it hints at interactivity that's not available

I'm not sure I understand this one. Could you share a screenshot?

Comment (VLuisa) [reply src/components/context-menu/context-menu.tsx:34]: @tristen yep! I don't think the disabled item should have a background color on hover
image

File: src/components/context-menu/context-menu.tsx (hunk 1 above)
Comment (VLuisa) [reply src/components/context-menu/context-menu.tsx:15]: probably? may be something we have to play with as we implement it in Studio.

File: src/docs/index.js (hunk 4 above)
Comment (VLuisa) [reply src/docs/index.js:5]: could be! are they reusable in other components for focus states?

**Comment by samanpwbb** [comment]
Wow

**Review by VLuisa — APPROVED** [review]
LGTM!

**Pull request merged**